dns/Dockerfile

```diff
@@ -0,0 +1,46 @@
+# syntax=docker/dockerfile:1
+
+ARG app_dir="/home/node/app"
+
+# * Base Node.js image
+FROM node:20-alpine AS base
+ARG app_dir
+WORKDIR ${app_dir}
+
+
+# * Installing production dependencies
+FROM base AS dependencies
+
+RUN --mount=type=bind,source=package.json,target=package.json \
+	--mount=type=bind,source=package-lock.json,target=package-lock.json \
+	--mount=type=cache,target=/root/.npm \
+	npm ci --omit=dev
+
+
+# * Installing development dependencies and building the application
+FROM base AS build
+
+RUN --mount=type=bind,source=package.json,target=package.json \
+	--mount=type=bind,source=package-lock.json,target=package-lock.json \
+	--mount=type=cache,target=/root/.npm \
+	npm ci
+
+COPY . .
+RUN npm run build
+
+
+# * Running the final application
+FROM base AS final
+ARG app_dir
+
+RUN npm install -g pm2
+
+ENV NODE_ENV production
+USER node
+
+COPY package.json .
+
+COPY --from=dependencies ${app_dir}/node_modules ${app_dir}/node_modules
+COPY --from=build ${app_dir}/dist ${app_dir}/dist
+
+CMD ["pm2-runtime", "start", "."]
```

Comment:
Is there still a reason to use pm2 here? As nice as pm2 is to manage nodejs processes on a vps, it's kinda odd to have a process manager (pm2) wrapping our process manager (docker).

It would also prevent docker daemon from knowing if the nodejs app is dead (since pm2 always keeps running)

Reply:
Just noticed this was already discussed in discord dev channels, whoops. I'll keep the review comment open though as it still applies.